### ISSUE TITLE
arch/x86/kvm/svm/vmenter.S: no such instruction: 'rdpru'

### ISSUE THREAD

**Issue by silver-airship** [sev-step/sev-step#2]
maybe rdpru is an arm instruction
so i can't fix it by my own.

**Comment by its-luca** [comment]
Hey @silver-airship 

Thank your for your interest in sev step.

Rdpru is an AMD only instructions that was introduced as part of the Zen 2 Architecture. See e.g. [1] for detailed information
Are you  trying to compile this on a pre Zen 2 CPU? You can use `cpuid -1 | grep -i rdpru` to check if your CPU supports it. However, this should only be an issue when you try to execute the code, not when compiling it.

Maybe your compiler version does not know the instruction yet? (I had this issue at one point)
Could you create a file `rdpru.S` with the following content and try to compile it with `gcc -c rdpru.S` ?
``` assembly
.text
rdpru
```

[1] https://www.amd.com/system/files/TechDocs/24594.pdf

**Comment by silver-airship** [comment]
I check the /proc/cpuinfo in a zen3 computer, but no rdpru flag.
So rdpru is only supported in zen2 arch?
If so, is it possible that use sev-step in zen3 arch?
How can i modify vmenter.S?
Thank you very much for your enthusiastic help！

**Comment by silver-airship** [comment]
I'm sure that zen3 arch also support rdpru!
The problem must lead by the compiler.
Thank you for your reply!

**Comment by silver-airship** [comment]
I update my gcc to gcc-11, but gcc-11 still can't recognize rdpru instruction.
Maybe you use aocc to compile it?
Or maybe i can modify it to `.byte 0x0f, 0x01, 0xfd`
This will work successfully, right?

**Comment by its-luca** [comment]
Hm this is weird. Just to narrow down the errors sources: Did you try to just compile the simple code snippet from my previous post?
As documented in the README, I developed SEV Step on Ubuntu 20.04.5 LTS which ships GCC 10.3. However, I have updated the system in the meantime. I just successfully compiled the snippet using `gcc (Ubuntu 11.3.0-1ubuntu1~22.04)`

Your suggestion, to replace the instruction with its raw byte sequence should work as well.
